src/EFCore.PG/Query/Internal/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -470,6 +508,50 @@ protected override Expression VisitNew(NewExpression newExpression)
                 : QueryCompilationContext.NotTranslatedExpression;
         }
 
+        // Translate new NpgsqlCube(...) -> cube(...)
+        if (newExpression.Constructor?.DeclaringType == typeof(NpgsqlCube))
+        {
+            if (!TryTranslateArguments(out var sqlArguments))
+            {
+                return QueryCompilationContext.NotTranslatedExpression;
+            }
+
+            var cubeTypeMapping = _typeMappingSource.FindMapping(typeof(NpgsqlCube));
+            var cubeParameters = newExpression.Constructor.GetParameters();
+
+            // Distinguish constructor overloads by parameter patterns
+            switch (cubeParameters)
+            {
+                case [var pCoords] when pCoords.ParameterType == typeof(double)
```

Comment:
Looks great, thanks for refactoring! One tiny nit: it's common to use IsAssignableFrom rather than IsAssignableTo nowadays, reversing the order and making things generally easier to read.